Fix Broken Packaging

### Context

Running `mvn -DskipTests pacakge` results in an error. Building `evosuite-master` fails with an `IllegalArgumentException`:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.1.0:shade (default) on project evosuite-master: Error creating shaded jar: null: IllegalArgumentException
```
Packaging still worked on the master branch. Likely, this issue was caused by 9d5849a0bb653967aa7476cfa14c37fff6d0d623 and 22622029afab8737e61da5223ec81320b5a7d0f9. These two commits bumped the Java language level to `11` to allow for automatic type inference in Java code using the `var` keyword. Related to this change, commit e8e57d2b0777ec40c2175ebaae4faca1c4238f86 bumped the `maven-plugin-plugin` version to 3.6.0.

### Steps to Reproduce

1. Switch to the EvoSuite root directory
2. Make you're using JDK 11 (check using `java -version` and `javac -version`)
3. Execute `mvn -DskipTests pacakge`

### EvoSuite Arguments
None.

### Additional info
Full stack trace:
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for EvoSuite 1.0.7-SNAPSHOT:
[INFO] 
[INFO] EvoSuite ........................................... SUCCESS [  0.296 s]
[INFO] evosuite-runtime ................................... SUCCESS [  0.489 s]
[INFO] evosuite-client .................................... SUCCESS [  0.352 s]
[INFO] evosuite-master .................................... FAILURE [ 29.669 s]
[INFO] evosuite-plugins ................................... SKIPPED
[INFO] Maven Plugin for EvoSuite .......................... SKIPPED
[INFO] evosuite-maven-plugin-test ......................... SKIPPED
[INFO] build-support-test ................................. SKIPPED
[INFO] evosuite-standalone-runtime ........................ SKIPPED
[INFO] evosuite-shaded .................................... SKIPPED
[INFO] evosuite-generated ................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  31.516 s
[INFO] Finished at: 2020-08-11T17:26:47+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.1.0:shade (default) on project evosuite-master: Error creating shaded jar: null: IllegalArgumentException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.1.0:shade (default) on project evosuite-master: Error creating shaded jar: null
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error creating shaded jar: null
    at org.apache.maven.plugins.shade.mojo.ShadeMojo.execute (ShadeMojo.java:546)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.IllegalArgumentException
    at org.objectweb.asm.ClassReader.<init> (Unknown Source)
    at org.objectweb.asm.ClassReader.<init> (Unknown Source)
    at org.objectweb.asm.ClassReader.<init> (Unknown Source)
    at org.apache.maven.plugins.shade.DefaultShader.addRemappedClass (DefaultShader.java:427)
    at org.apache.maven.plugins.shade.DefaultShader.shadeSingleJar (DefaultShader.java:229)
    at org.apache.maven.plugins.shade.DefaultShader.shadeJars (DefaultShader.java:190)
    at org.apache.maven.plugins.shade.DefaultShader.shade (DefaultShader.java:106)
    at org.apache.maven.plugins.shade.mojo.ShadeMojo.execute (ShadeMojo.java:442)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :evosuite-master
```

--- a/client/src/main/java/org/evosuite/ga/metaheuristics/TestSuiteAdapter.java
+++ b/client/src/main/java/org/evosuite/ga/metaheuristics/TestSuiteAdapter.java
@@ -331,7 +331,7 @@ public abstract class TestSuiteAdapter<A extends GeneticAlgorithm<TestChromosome
             throws IllegalArgumentException {
         if (algorithm != null) {
             if (factory instanceof TestSuiteChromosomeFactoryMock) {
-                final var tcfw = (TestSuiteChromosomeFactoryMock) factory;
+                TestSuiteChromosomeFactoryMock tcfw = (TestSuiteChromosomeFactoryMock) factory;
                 algorithm.setChromosomeFactory(tcfw.getWrapped());
             } else {
                 throw new IllegalArgumentException("factory not supported: " + factory);

--- a/client/src/main/java/org/evosuite/ga/metaheuristics/mosa/MOSA.java
+++ b/client/src/main/java/org/evosuite/ga/metaheuristics/mosa/MOSA.java
@@ -29,6 +29,7 @@ import org.evosuite.ga.operators.selection.RandomKSelection;
 import org.evosuite.ga.operators.selection.RankSelection;
 import org.evosuite.ga.operators.selection.SelectionFunction;
 import org.evosuite.rmi.ClientServices;
+import org.evosuite.rmi.service.ClientNodeLocal;
 import org.evosuite.statistics.RuntimeVariable;
 import org.evosuite.testcase.TestChromosome;
 import org.evosuite.testcase.TestFitnessFunction;
@@ -171,7 +172,8 @@ public class MOSA extends AbstractMOSA {
 			this.distance.fastEpsilonDominanceAssignment(this.rankingFunction.getSubfront(i), this.getUncoveredGoals());
 		}
 
-		final var clientNode = ClientServices.<TestChromosome>getInstance().getClientNode();
+		final ClientNodeLocal<TestChromosome> clientNode =
+				ClientServices.<TestChromosome>getInstance().getClientNode();
 
 		Listener<Set<TestChromosome>> listener = null;
 		if (Properties.NUM_PARALLEL_CLIENTS > 1) {

--- a/client/src/main/java/org/evosuite/strategy/PropertiesSuiteGAFactory.java
+++ b/client/src/main/java/org/evosuite/strategy/PropertiesSuiteGAFactory.java
@@ -62,7 +62,6 @@ import org.evosuite.utils.ArrayUtil;
 import org.evosuite.utils.ResourceController;
 import sun.misc.Signal;
 
-
 /**
  * Factory for GA on test suites
  * 
@@ -79,7 +78,8 @@ public class PropertiesSuiteGAFactory
 			switch (Properties.TEST_FACTORY) {
 			case ALLMETHODS:
 				logger.info("Using all methods chromosome factory");
-				return new TestSuiteChromosomeFactory(new AllMethodsTestChromosomeFactory());
+				return new TestSuiteChromosomeFactory(
+				        new AllMethodsTestChromosomeFactory());
 			case RANDOM:
 				logger.info("Using random chromosome factory");
 				return new TestSuiteChromosomeFactory(new RandomLengthTestFactory());
@@ -88,11 +88,13 @@ public class PropertiesSuiteGAFactory
 				return new TestSuiteChromosomeFactory(new ArchiveTestChromosomeFactory());
 			case JUNIT:
 				logger.info("Using seeding chromosome factory");
-				final var factory = new JUnitTestCarvedChromosomeFactory(new RandomLengthTestFactory());
+				JUnitTestCarvedChromosomeFactory factory = new JUnitTestCarvedChromosomeFactory(
+				        new RandomLengthTestFactory());
 				return new TestSuiteChromosomeFactory(factory);
             case SERIALIZATION:
                 logger.info("Using serialization seeding chromosome factory");
-                return new SerializationSuiteChromosomeFactory(new RandomLengthTestFactory());
+                return new SerializationSuiteChromosomeFactory(
+                        new RandomLengthTestFactory());
 			default:
 				throw new RuntimeException("Unsupported test factory: "
 				        + Properties.TEST_FACTORY);
@@ -125,7 +127,7 @@ public class PropertiesSuiteGAFactory
 				return new MuLambdaEA<>(factory, Properties.MU, Properties.LAMBDA);
 			case MONOTONIC_GA: {
 				logger.info("Chosen search algorithm: MonotonicGA");
-				final var ga = new MonotonicGA<>(factory);
+				MonotonicGA<TestSuiteChromosome> ga = new MonotonicGA<>(factory);
 				if (Properties.REPLACEMENT_FUNCTION == TheReplacementFunction.FITNESSREPLACEMENT) {
 					// user has explicitly asked for this replacement function
 					ga.setReplacementFunction(new FitnessReplacementFunction<>());
@@ -137,7 +139,7 @@ public class PropertiesSuiteGAFactory
 			}
 			case CELLULAR_GA: {
 				logger.info("Chosen search algorithm: CellularGA");
-				final var ga = new CellularGA<>(Properties.MODEL,	factory);
+				CellularGA<TestSuiteChromosome> ga = new CellularGA<>(Properties.MODEL,	factory);
 				if (Properties.REPLACEMENT_FUNCTION == TheReplacementFunction.FITNESSREPLACEMENT) {
 					// user has explicitly asked for this replacement function
 					ga.setReplacementFunction(new FitnessReplacementFunction<>());
@@ -150,7 +152,7 @@ public class PropertiesSuiteGAFactory
 			case STEADY_STATE_GA: {
 			logger.info("Chosen search algorithm: Steady-StateGA");
 				logger.info("Chosen search algorithm: Steady-StateGA");
-				final var ga = new SteadyStateGA<>(factory);
+				SteadyStateGA<TestSuiteChromosome> ga = new SteadyStateGA<>(factory);
 				if (Properties.REPLACEMENT_FUNCTION == TheReplacementFunction.FITNESSREPLACEMENT) {
 					// user has explicitly asked for this replacement function
 					ga.setReplacementFunction(new FitnessReplacementFunction<>());
@@ -283,24 +285,24 @@ public class PropertiesSuiteGAFactory
 
 	@Override
 	public GeneticAlgorithm<TestSuiteChromosome> getSearchAlgorithm() {
-		final var factory = getChromosomeFactory();
+		ChromosomeFactory<TestSuiteChromosome> factory = getChromosomeFactory();
 		
 		// FIXXME
-		final var ga = getGeneticAlgorithm(factory);
+		GeneticAlgorithm<TestSuiteChromosome> ga = getGeneticAlgorithm(factory);
 
 		if (Properties.NEW_STATISTICS)
 			ga.addListener(new StatisticsListener<>());
 
 		// How to select candidates for reproduction
-		final var selectionFunction = getSelectionFunction();
+		SelectionFunction<TestSuiteChromosome> selectionFunction = getSelectionFunction();
 		selectionFunction.setMaximize(false);
 		ga.setSelectionFunction(selectionFunction);
 
-		final var ranking_function = getRankingFunction();
+		RankingFunction<TestSuiteChromosome> ranking_function = getRankingFunction();
 		ga.setRankingFunction(ranking_function);
 
 		// When to stop the search
-		final var stopping_condition = getStoppingCondition();
+		StoppingCondition<TestSuiteChromosome> stopping_condition = getStoppingCondition();
 		ga.setStoppingCondition(stopping_condition);
 		// ga.addListener(stopping_condition);
 		if (Properties.STOP_ZERO) {
@@ -325,7 +327,7 @@ public class PropertiesSuiteGAFactory
 		ga.setPopulationLimit(getPopulationLimit());
 
 		// How to cross over
-		final var crossover_function = getCrossoverFunction();
+		CrossOverFunction<TestSuiteChromosome> crossover_function = getCrossoverFunction();
 		ga.setCrossOverFunction(crossover_function);
 
 		// What to do about bloat
@@ -333,7 +335,8 @@ public class PropertiesSuiteGAFactory
 		// ga.setBloatControl(bloat_control);
 
 		if (Properties.CHECK_BEST_LENGTH) {
-			final var bloat_control = new RelativeSuiteLengthBloatControl<TestSuiteChromosome>();
+			RelativeSuiteLengthBloatControl<TestSuiteChromosome> bloat_control =
+					new RelativeSuiteLengthBloatControl<>();
 			ga.addBloatControl(bloat_control);
 			ga.addListener(bloat_control);
 		}
@@ -367,13 +370,13 @@ public class PropertiesSuiteGAFactory
 		if (Properties.SHUTDOWN_HOOK) {
 			// ShutdownTestWriter writer = new
 			// ShutdownTestWriter(Thread.currentThread());
-			final var writer = new ShutdownTestWriter<TestSuiteChromosome>();
+			ShutdownTestWriter<TestSuiteChromosome> writer = new ShutdownTestWriter<>();
 			ga.addStoppingCondition(writer);
 			RMIStoppingCondition<TestSuiteChromosome> rmi = RMIStoppingCondition.getInstance();
 			ga.addStoppingCondition(rmi);
 
 			if (Properties.STOPPING_PORT != -1) {
-				final var ss = new SocketStoppingCondition<TestSuiteChromosome>();
+				SocketStoppingCondition<TestSuiteChromosome> ss = new SocketStoppingCondition<>();
 				ss.accept();
 				ga.addStoppingCondition(ss);
 			}

--- a/client/src/main/java/org/evosuite/testcase/TestChromosome.java
+++ b/client/src/main/java/org/evosuite/testcase/TestChromosome.java
@@ -276,7 +276,8 @@ public final class TestChromosome extends AbstractTestChromosome<TestChromosome>
 	 * @param objective*/
 	@Override
 	public boolean localSearch(LocalSearchObjective<TestChromosome> objective) {
-		final var localSearch = TestCaseLocalSearch.selectTestCaseLocalSearch();
+		TestCaseLocalSearch<TestChromosome> localSearch =
+				TestCaseLocalSearch.selectTestCaseLocalSearch();
 		return localSearch.doSearch(this, objective);
 	}
 

--- a/plugins/maven/pom.xml
+++ b/plugins/maven/pom.xml
@@ -91,7 +91,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-plugin-plugin</artifactId>
-                    <version>3.6.0</version>
+                    <version>3.5.1</version>
                     <configuration>
                         <!-- see http://jira.codehaus.org/browse/MNG-5346 -->
                         <skipErrorNoDescriptorsFound>true</skipErrorNoDescriptorsFound>

--- a/pom.xml
+++ b/pom.xml
@@ -247,8 +247,8 @@
                 <jdk>[9,)</jdk>
             </activation>
             <properties>
-                <compiler.target>11</compiler.target>
-                <compiler.source>11</compiler.source>
+                <compiler.target>1.8</compiler.target>
+                <compiler.source>1.8</compiler.source>
                 <exclude.tests.concolic>**/*Concolic*.java</exclude.tests.concolic>
                 <exclude.tests.mimeType>**/*TestMIMEType.java</exclude.tests.mimeType>
             </properties>

